op-service/eth/transactions.go

```diff
@@ -115,3 +115,28 @@ func CheckRecentTxs(
 	}
 	return oldestBlock.Uint64(), false, nil
 }
+
+type GenericTx interface {
+	// Transaction tries to interpret into a typed tx.
+	// This will return types.ErrTxTypeNotSupported if the transaction-type is not supported.
+	Transaction() (*types.Transaction, error)
+
+	// TxType returns the EIP-2718 type.
+	TxType() uint8
```

Comment:
We should consider returning an error from this method, in case the transaction cannot be decoded as an EIP-2718 transaction (example: first byte is outside of the specified range).